malformed/test_chunking_example.py

```diff
@@ -0,0 +1,235 @@
+from framework import tester
+from helpers import tf_cfg, deproxy
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2022 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+# This small testsuite is created to serve as an example or a template for
+# tests with iteration over various chunk sizes. It demonstrates how to do
+# this job in the right way.
+#
+# Every test is splitted into two functions: inner_test_xxx() and test_xxx().
+# Then the inner_test_xxx() is iterated by the 'universal' function
+# iterate_test(), whereas outer text_xxx() is responsible for start and
+# setting required parameters, if any.
+#
+# Three example tests are defined here:
+# 1. Test for forwarding the request with long and sophisticated header set.
+# 2. A simple request test for chunked response from the backend server.
+# 3. A simple request test transmitted over TLS.
+
+class CorrectHeadersTest(tester.TempestaTest):
+
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' :
+"""HTTP/1.1 200 OK
+Content-Length: 0
+Connection: keep-alive
+
+"""
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+cache 0;
+server ${general_ip}:8000;
+listen 80;
+listen 443 proto=https;
+tls_certificate ${general_workdir}/tempesta.crt;
+tls_certificate_key ${general_workdir}/tempesta.key;
+""",
+    }
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+        {
+            'id' : 'deproxy_ssl',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '443',
+            'ssl'  : True
+        },
+    ]
+
+    def iterate_test(self, test_func, request, *args, **kwargs):
+        # This function provides iterations over various chunk sizes.
+        # It have required positional parameters:
+        # @test_func - a function to iterate. Supposed, it implements
+        #              a test scenario and is a member of the same class.
+        # @request - a request to be sent
+        # The function can accept additional positional and keyword
+        # parameters with *args and **kwargs to forward them to
+        # the @test_func().
+        CHUNK_SIZES = [ 1, 2, 3, 4, 8, 16, 32, 64, 128, 256, 1500, 9216,
+                        1024*1024 ]
+        for i in range(len(CHUNK_SIZES)):
+            test_func(CHUNK_SIZES[i], request, *args, **kwargs)
+            if CHUNK_SIZES[i] > len(request):
+                break;
```

Comment:
can be simplified
```Python
for c in CHUNK_SIZES:
    test_func(c, request, *args, **kwargs)
    if c > len(request):
        break
```

Reply:
OK, probably a generator won't simplify the code much - we still need the sequence and check the chunk size against the request length. So no need to do anything with https://github.com/tempesta-tech/tempesta-test/pull/195#discussion_r776091652